### ISSUE TITLE
[BUG] Users cannot interact with the PR checklist

### ISSUE THREAD

**Issue by Remi-Gau** [nilearn/nilearn#3487]
Not a nilearn bug, but more of nilearn repo UX issue.

It seems that it is not possible for "some" (new?) users to actually use the PR checklist to tick off items to make sure they comply with the contributing guidelines.

### Expected behavior

For users to be able to tick off items of the PR checklist.

### Actual behavior

No matter how much clicking I do, nothing happens.

This is not an issue on PRs opened on my fork: https://github.com/Remi-Gau/nilearn/pull/2#issuecomment-1416690220

### Steps and code to reproduce bug

This may just be me, so I would be glad to have someone prove me wrong and show me this is a [PEBKAC error
](https://slang.net/meaning/pebcac).

Unsure how best to reproduce this, unless maybe having some other users who do not have triage access to the repo.

Any thoughts?

**Comment by bthirion** [comment]
No solution envisioned btw my chair and my computer.

**Comment by Remi-Gau** [comment]
Can try to see if using this action could help edit append the checklist to the top comment that should be editable by the person who opened the PR. 

https://github.com/marketplace/actions/sticky-pull-request-comment

**Comment by ymzayek** [comment]
@Remi-Gau indeed I can edit anyone's checklist. So it must be a permissions issue where contributors cannot edit the bot's comment. Once we merge your first PR we can see if it's just blocking first time contributors or more generally anyone that is not included as a maintainer. But in any case that action version needs to be updated to this one: https://github.com/marketplace/actions/auto-comment. Or we can use this new one you suggested. Maybe you can demo what that would look like?

**Comment by Remi-Gau** [comment]
Yup I will most likely play around on my fork or in a sandbox repo before opening a PR.

**Comment by Remi-Gau** [comment]
By the way do you want me to set up dependabot to automatically open pr to update github actions to their latest versions?

**Comment by ymzayek** [comment]
Yea we can try that. Those things usually go unnoticed and eventually break so better to be ahead of it.

**Comment by Remi-Gau** [comment]
> Once we merge your first PR we can see if it's just blocking first time contributors or more generally anyone that is not included as a maintainer.

I confirm that even once my first PR is merged I cannot edit new PR checklists.

**Comment by emdupre** [comment]
I will say that this is an issue that JOSS has been dealing with for some time on the reviewer checklists, and the solution(s) have been somewhat complex logistically -- so I'm not excited about porting them over. 

Originally, every reviewer had to be invited to be part of the openjournals (in our case, the Nilearn) GitHub organization. Then they would have edit access to check each box in their review.

The second (current) solution is to use the github bot associated with the openjournals org (`editorialbot`) to automatically generate reviewer checklists on demand.

At this point : I'm really not confident that these checklists will be used often enough to warrant the work required to make them checkable by the submitter. As a first pass, maybe we should have PR reviewers start using them to decide if we're happy with the structure ?

**Comment by Remi-Gau** [comment]
> I'm really not confident that these checklists will be used often enough to warrant the work required to make them checkable by the submitter. 

Given how often I have seen people ignore issue forms to open unstructured blank issues, I am unfortunately inclined to agree with you. 

> As a first pass, maybe we should have PR reviewers start using them to decide if we're happy with the structure ?

That and automate as CI things that can be: though beyond the PEP8 compliance I am not sure which other item can automated in a meaningful way.

**Comment by Remi-Gau** [comment]
Will close this for now.

I think I have grown to view the PR checklist as something that the code reviewers / maintainers are filling in and not something to be filled by the person who opened the PR

**Comment by Remi-Gau** [comment]
came across this action that maybe could be used if we want to revisit this issue:

https://github.com/peter-evans/create-or-update-comment
